### ISSUE TITLE
Which files will need to be updated?

### ISSUE THREAD

**Issue by ali6836** [dividuum/fastmc#6]
Which files will need to be updated for this project? I am planning on possibly updating it myself, but which files will need to be updated/contain protocols

**Comment by dividuum** [comment]
That would be [proto.py](https://github.com/dividuum/fastmc/blob/master/fastmc/proto.py). Updating would require you to include all changes made since 1.8 according to the protocol documentation at http://wiki.vg/Protocol_History. It would probably also require that your write custom decoders for new primitives (maybe? I didn't check to see if there is anything new). It's quite fiddly and no easy task.

**Comment by ali6836** [comment]
Seems quite complex, would it even be possible for an absolute noob at coding like me? Also, what is a primitive?

**Comment by dividuum** [comment]
To be honest: Probably not. It requires quite a bit of bit-fiddling (in case there are new data types) and a solid sense of how network protocols work and how to debug them in case the decoding goes wrong somewhere. If you have no experience with that, you're in for a rough ride.

**Comment by ali6836** [comment]
Ah right. Well I probably won't do anything then. Sorry for wasting your time, and hopefully you will possibly update it in the near future.

**Comment by dividuum** [comment]
Don't worry. No time wasted :-)

I hope to find the time to update it at some point. But I haven't touched Minecraft in a while (:sadface:), so motivation is a bit low at the moment.

**Comment by ali6836** [comment]
I have found this, and am wondering whether or not this is similar to what needs to be updated. It is on the latest version of minecraft, I am just wondering whether it will be compatible, or is similar. https://github.com/ammaraskar/pyCraft/tree/master/minecraft

**Comment by dividuum** [comment]
Not compatible. The API seems fairly different. But it's looks like it is able to parse newer version of the protocol. Nice find.

**Comment by ali6836** [comment]
Well, since it does include all of the newer protocols, would it be possible to change the format of protocol that fastmc reads?

**Comment by dividuum** [comment]
Not really. It might help if things differ from the documentation on wiki.vg, but that's all. I can't use it directly. Its approach to parsing it quite a bit different.
